src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -1352,6 +1354,15 @@ protected RubyMethod method(VirtualFrame frame, Object self, Object name,
                 @Cached ConditionProfile notFoundProfile,
                 @Cached ConditionProfile respondToMissingProfile,
                 @Cached LogicalClassNode logicalClassNode) {
+            final Frame callerFrame = getContext()
+                    .getCallStack()
+                    .getCallerFrameIgnoringSend(FrameInstance.FrameAccess.READ_ONLY);
```

Comment:
Kernel#method should be fast and basically just be an inline cache on the instance class + name.
This causes to iterate the stack every time.

Let's use the same approach as in RespondToNode.

Reply:
Hm, but `ReadCallerFrameNode` (used in `RespondToNode`) also iterate the stack, isn't? 
https://github.com/oracle/truffleruby/blob/9da7552b00072a6c06a0cb1e1a2cce4c9da0bdaa/src/main/java/org/truffleruby/language/arguments/ReadCallerDataNode.java#L31-L39

It just has a condition profiler to "skip" an iteration sometimes (not sure how often)

Reply:
The first branch is basically always taken after the first call for a given call site, so it only iterates once per call site, and then all later calls just read the caller frame from the arguments of the current frame (except edge cases like a non-Ruby caller, but we can detect those by being passed a `null` frame or so).